shared/chat/conversation/list.desktop.js

```diff
@@ -47,14 +49,28 @@ class ConversationList extends Component<void, Props, State> {
       // minus one because loader message is there
       const messageIndex = index - 1
       const message = this.state.messages.get(messageIndex)
-      const id = message && message.messageID
+      // We want a stable key -- messages have an outboxID but no messageID,
+      // then later gain a messageID.  So if we prefer outboxIDs to messageIDs
+      // for the key, every row keeps its key.
+      const id = message && (message.outboxID || message.messageID)
       if (id == null) {
         console.warn('id is null for index:', messageIndex)
       }
       return id
     }
   }
 
+  componentWillUpdate (nextProps: Props, nextState: State) {
+    // If a message has moved from pending to sent, tell the List to discard
+    // heights for it and everything after it, so that they're re-rendered.
```

Comment:
:sheep: This comment confused me for a few minutes, because I asked myself what code caused "everything after it" to be discarded. I think this just explains what `recomputeRowHeights` does? If so, perhaps worth simplifying the wording to avoid this confusion?